py/mpconfig.h

```diff
@@ -644,6 +644,16 @@ typedef double mp_float_t;
 #define MP_PLAT_FREE_EXEC(ptr, size) m_del(byte, ptr, size)
 #endif
 
+// Cast mp_obj_t to object pointer
+#ifndef MP_OBJ_CAST
```

Comment:
> but rather for uPy itself to configure based on what object model is being used.

... which is itself might be configured by port. Of course, there's a long way to GC method being configurable per port, though that can be start.

Other reason I put it to mpconfig.h was that it's wrapped in #ifndef. Are you ok to have them in obj.h wrapped like that?